src/filters/detectors/assurance.cpp

```diff
@@ -9,71 +10,174 @@
 namespace mesos {
 namespace serenity {
 
-Result<Detection> AssuranceDetector::processSample(
-    double_t in) {
-  this->window.push_back(in);
+void AssuranceDetector::shiftBasePoints() {
```

Comment:
You have everything here: https://github.com/mesosphere/serenity/pull/107/files#diff-7e2fcb5059d36af5d2d9838aa5b0cf16R81